src/mumble/AudioOutput.cpp

```diff
@@ -168,35 +183,69 @@ void AudioOutput::addFrameToBuffer(ClientUser *sender, const Mumble::Protocol::A
 
 		qrwlOutputs.lockForWrite();
 
-		aop = new AudioOutputSpeech(sender, iMixerFreq, audioData.usedCodec, iBufferSize);
-		qmOutputs.replace(sender, aop);
+		speech = new AudioOutputSpeech(sender, iMixerFreq, audioData.usedCodec, iBufferSize);
+		qmOutputs.replace(sender, speech);
 	}
 
-	aop->addFrameToBuffer(audioData);
+	speech->addFrameToBuffer(audioData);
 
 	qrwlOutputs.unlock();
 }
 
-void AudioOutput::removeBuffer(const ClientUser *user) {
-	removeBuffer(qmOutputs.value(user));
-}
-
-void AudioOutput::removeBuffer(AudioOutputUser *aop) {
+void AudioOutput::handleInvalidatedBuffer(AudioOutputBuffer *buffer) {
 	QWriteLocker locker(&qrwlOutputs);
-	QMultiHash< const ClientUser *, AudioOutputUser * >::iterator i;
+	QMultiHash< const ClientUser *, AudioOutputBuffer * >::iterator i;
 	for (i = qmOutputs.begin(); i != qmOutputs.end(); ++i) {
```

Comment:
```suggestion
	for (auto iter = qmOutputs.begin(); iter != qmOutputs.end(); ++iter) {
```
one of the very few exceptions where I actually think the `auto` keyword is useful: for iterators